include/seqan3/search/fm_index/fm_index_cursor.hpp

```diff
@@ -32,6 +32,45 @@ namespace seqan3
  * \{
  */
 
+//!\brief The underlying suffix array intervals.
+struct suffix_array_interval
+{
+    //!\brief The begin position of the interval ("left boundary").
+    size_t begin_position;
+    //!\brief The exclusive end position of the interval ("right boundary").
+    size_t end_position;
+
+    //!\brief Convert to tuple.
+    operator std::tuple<size_t &, size_t &>()
```

Comment:
```suggestion
    operator std::tuple<size_t, size_t>() const noexcept
```
So this wouldn't work?

Because your version should require that this is an lvalue reference as well. Then the question remains, what is with the other versions `const &` should be also added. Move version are not required since we move/copy built-in types anyway.

Reply:
Nope https://godbolt.org/z/j9avh9